testsuite/tests/libratest/smoke_test.rs

```diff
@@ -1174,3 +1185,65 @@ fn test_malformed_script() {
         .mint_coins(&["mintb", "0", "10"], true)
         .unwrap();
 }
+
+#[test]
+fn test_key_manager_consensus_rotation() {
+    // Create and launch a local validator swarm of 2 nodes.
+    let mut swarm = TestEnvironment::new(2);
+    swarm.launch_swarm(RoleType::Validator);
+
+    // Create a node config for the key manager by modifying the first node config in the swarm.
```

Comment:
we should probably set the ability to add a template to TestEnvironment::new(num_validators, template). I think the reason we didn't do it in the past was because there was only one use case... now we have another....

Reply:
Thanks! I've added a TODO for this (for now). The problem is that passing a template into the TestEnvironment::new() call doesn't solve the issue entirely, because we still need to update the json rpc port for the key manager config (which can only be done once the config for each swarm node has already been generated -- as they are random ports). I might return to this once I have spare cycles